File: frontend/web/features/management/types/student.ts
export type StudentValues = Record<string, string> & {
  photoDataUrl: string;
  number: string;
  name: string;
  email: string;
  departmentId: string;
  department: string;
  year: string;
  status: string;
};

export type StudentItem = { id: string; values: StudentValues };

File: backend/src/InstituteManagement.Domain/Entities/Student.cs
namespace InstituteManagement.Domain.Entities;

public sealed class Student : Entity
{
    public required string StudentCode { get; set; }
    public required string FullName { get; set; }
    public string Email { get; set; } = string.Empty;
    public string PhotoDataUrl { get; set; } = string.Empty;
    public Guid DepartmentId { get; set; }
    public Department? Department { get; set; }
    public int YearLevel { get; set; }
    public string Shift { get; set; } = "Morning";
    public string Status { get; set; } = "Active";
}
